apps/balancer-tools/src/app/apr/layout.tsx

```diff
@@ -15,12 +16,14 @@ export const metadata: Metadata = {
 export default function Layout({ children }: { children: React.ReactNode }) {
   return (
     <div className="flex h-full flex-col">
-      <Header
-        linkUrl={"/apr"}
-        title={APP_DISPLAY_NAME}
-        imageSrc={"/assets/balancer-symbol.svg"}
-        endButton={<HeaderEndButton />}
-      />
+      <React.Suspense fallback={<Spinner />}>
+        <Header
+          linkUrl={"/apr"}
+          title={APP_DISPLAY_NAME}
+          imageSrc={"/assets/balancer-symbol.svg"}
+          endButton={<HeaderEndButton />}
+        />
+      </React.Suspense>
```

Comment:
😮

Reply:
just curious about how you found this out, and what part of this component was throwing an error/loading external data?

Reply:
`HeaderEndButton` has `const searchParams = useSearchParams();` that's why the component needs a suspense

Reply:
so the suspense should be only within the HeaderEndButton, no? 👁️ 👄 👁️

Reply:
heheheheh true